### PR TITLE
WW-5196 use generics for RequestMap and ApplicationMap and correct SessionMap to also be of type <String, Object>

### PR DESCRIPTION
**Pull request by sdutry**
Assumption: 
- keys will always be Strings

Extracted abstract class for similar logic for entries

Resolves [WW-5196](https://issues.apache.org/jira/browse/WW-5196)

### CODE REVIEW

**Comment by lukaszlenart** [comment]
Nice 👏 but I would wait with merging this till 6.0.1 is out.

**Comment by sdutry** [comment]
Sure, no problem with waiting, just wondering if i should adjust SessionMap accordingly since, as far as i can see, it should only have String keys too.

**Comment by lukaszlenart** [comment]
> just wondering if i should adjust SessionMap accordingly since, as far as i can see, it should only have String keys too.

Yes, please - adjust description of the ticket as well, thanks a lot!

File: core/src/main/java/org/apache/struts2/dispatcher/ApplicationMap.java [hunk 1]
```diff
@@ -167,24 +128,26 @@ public Object get(Object key) {
      * @param value the value to set.
      * @return the attribute that was just set.
      */
-    public Object put(Object key, Object value) {
-        Object oldValue = get(key);
-        entries = null;
-        context.setAttribute(key.toString(), value);
-        return oldValue;
-    }
+	public Object put(final String key, final Object value) {
```
Comment (sepe81) [inline core/src/main/java/org/apache/struts2/dispatcher/ApplicationMap.java:131]: fix tabs to spaces for method put

File: core/src/main/java/org/apache/struts2/dispatcher/StringObjectEntry.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package org.apache.struts2.dispatcher;
+
+import java.util.Map.Entry;
+
+abstract class StringObjectEntry implements Entry<String, Object> {
+    private String key;
+    private Object value;
+
+    public StringObjectEntry(final String key, final Object value) {
```
Comment (sepe81) [inline core/src/main/java/org/apache/struts2/dispatcher/StringObjectEntry.java:9]:
```suggestion
    StringObjectEntry(final String key, final Object value) {
```

see sonar rule java:S5993

**Review by sepe81 — CHANGES_REQUESTED** [review]
some small changes

**Comment by kusalk** [comment]
LGTM

**Comment by lukaszlenart** [comment]
@sdutry could rebase this PR? We switched from Travis to GH Actions to validate PRs, thanks in advance!

**Comment by sdutry** [comment]
@lukaszlenart i'll der what i can do and if it's still relevant when i find a moment

**Comment by kusalk** [comment]
@sdutry Definitely still relevant as I was about to put up a PR for the same but then noticed yours. If you've no time I'd be happy to rebase it for you :)

**Comment by sdutry** [comment]
I'll try and do it tonight

**Comment by sdutry** [comment]
@lukaszlenart tried to merge in master to get it up to date again.

I had to change a test concerning the nonce in order to get that test running again, not comfortable with having to do a change there, don't think i touched anything there. (`FreemarkerResultMockedTest.testNonce`)

Can not get the java 11 build succeeding due to the following error :
```
Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar (default-cli) on project struts2-parent: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator
```

**Comment by lukaszlenart** [comment]
> Can not get the java 11 build succeeding due to the following error :
> 
> ```
> Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar (default-cli) on project struts2-parent: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator
> ```

Thanks a lot @sdutry the issue with JDK11 is related to GH Action token access and can be ignored.

**Review by lukaszlenart — APPROVED** [review]
LGTM 👍

**Pull request merged**